core/shared/src/test/scala/fs2/concurrent/ChannelSuite.scala

```diff
@@ -251,4 +251,28 @@ class ChannelSuite extends Fs2Suite {
 
     TestControl.executeEmbed(test)
   }
+
+  test("sendAndClose closes right after sending the last element") {
+    val result = Channel.bounded[IO, Int](1).flatMap { ch =>
+      ch.sendAndClose(0) *> ch.isClosed
```

Comment:
Would it make sense to have this test for the unbounded Channel case as well, similarly to the "racing" scenarios? :slightly_smiling_face: 
Also, would it be a good idea to cover both impacted branches in the Channel implementation (`size < capacity` and `else`)? :slightly_smiling_face: